### PR TITLE
PN-90: Genotypic seeding results need to be combined with phenotypic seeding.

### PR DESCRIPTION
**Pull request by veronikaslc**
Fixed.

### CODE REVIEW

File: similarity-data-impl/src/test/java/org/phenotips/data/similarity/internal/RestrictedPatientGenotypeSimilarityViewTest.java [hunk 1]
```diff
@@ -457,25 +455,25 @@ public void testNoVariantsInCandidateGene()
         Assert.assertEquals(2, results.size());
 
         JSONObject top = results.getJSONObject(0);
-        Assert.assertTrue(top.getString("gene").equals("SRCAP"));
-        Assert.assertTrue(top.getDouble("score") > 0.9);
+        Assert.assertTrue(top.getString("gene").equals("NOTCH2"));
+        Assert.assertTrue(top.getDouble("score") > 0.5);
 
         // Ensure match shows underlying variant details
```
Comment (buske) [inline similarity-data-impl/src/test/java/org/phenotips/data/similarity/internal/RestrictedPatientGenotypeSimilarityViewTest.java]: Comment isn't accurate?

File: similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/DefaultPatientGenotypeSimilarityView.java [hunk 2]
```diff
@@ -105,7 +106,8 @@ private PatientGenotypeManager getGenotypeManager()
      */
     private void matchGenes()
     {
-        for (String gene : getGenes()) {
+        Set<String> genes = getGenes();
```
Comment (sdumitriu) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/DefaultPatientGenotypeSimilarityView.java]: Why this change?

Comment (veronikaslc) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/DefaultPatientGenotypeSimilarityView.java]: I guess, unnecessary one

File: similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java [hunk 3]
```diff
@@ -204,4 +211,41 @@ private long count(SolrQuery query)
...
+                // Treat empty status as candidate
+                if (StringUtils.isBlank(status) || "solved".equals(status) || "candidate".equals(status)) {
+                    genesToSearch.add(geneName);
+                }
+            }
+        }
+        if (!genesToSearch.isEmpty()) {
+            String geneQuery = getQueryFromTerms(genesToSearch);
+            q.append(" solved_genes: " + geneQuery);
+            q.append(" candidate_genes: " + geneQuery);
```
Comment (sdumitriu) [inline similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java]: If you leave a space between `:` and `(`, then that is no longer a field specifier, so the query will actually look for the text `solved_genes` and each individual gene name in the fields specified in the qf parameter.

File: similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractPatientGenotypeSimilarityView.java [hunk 4]
```diff
@@ -93,7 +93,7 @@ public boolean hasGenotypeData()
             Set<String> matchGenes = this.matchGenotype.getCandidateGenes();
             Set<String> refGenes = this.refGenotype.getCandidateGenes();
             sharedGenes.addAll(matchGenes);
-            sharedGenes.retainAll(refGenes);
+            sharedGenes.addAll(refGenes);
```
Comment (sdumitriu) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractPatientGenotypeSimilarityView.java]: This looks wrong. `sharedGenes` means genes that both patients have in common, not the union of their genes.

Comment (sdumitriu) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractPatientGenotypeSimilarityView.java]: If the meaning of the method changes, then its name should change as well, but that would be an API breakage.

Comment (buske) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractPatientGenotypeSimilarityView.java]: It was an intentional API breakage.

File: similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/DefaultPatientGenotypeSimilarityView.java (hunk 2 above)
Comment (buske) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/DefaultPatientGenotypeSimilarityView.java]: Indeed, likely a relic from my debugging. Sorry.
